packages/zoe/src/contractSupport/priceAuthorityInitial.js

```diff
@@ -0,0 +1,118 @@
+// @ts-check
+import { E } from '@endo/far';
+import { Far } from '@endo/marshal';
+
+import { makeNotifier } from '@agoric/notifier';
+import { AmountMath } from '@agoric/ertp';
+
+import { multiplyBy } from './ratio.js';
+
+/** @template T @typedef {import('@endo/eventual-send').EOnly<T>} EOnly */
+
+/**
+ * Override `makeQuoteNotifier`, `quoteGiven` to provide an initial price
+ * in case one is not yet available from the source.
+ *
+ * Mainly for testing vaults without waiting for oracle operators to PushPrice.
+ *
+ * @param {Ratio} initialPrice
+ * @param {PriceAuthority} priceAuthority
+ * @param {ERef<Mint<'set'>>} quoteMint
+ * @param {Brand<'nat'>} brandIn
+ * @param {Brand<'nat'>} brandOut
+ * @returns {PriceAuthority}
+ */
+export const makeInitialTransform = (
+  initialPrice,
+  priceAuthority,
+  quoteMint,
+  brandIn,
+  brandOut,
+) => {
+  assert.equal(initialPrice.numerator.brand, brandOut);
+  assert.equal(initialPrice.denominator.brand, brandIn);
+  let initialMode = true;
+
+  const quoteBrandP = E(E(quoteMint).getIssuer()).getBrand();
+  const timerP = E(priceAuthority).getTimerService(brandIn, brandOut);
+
+  /**
+   * @param {Amount<'nat'>} amountIn
+   * @param {Amount<'nat'>} amountOut
+   * @returns {Promise<PriceQuote>}
+   */
+  const oneQuote = async (amountIn, amountOut) => {
+    const [quoteBrand, timer, timestamp] = await Promise.all([
+      quoteBrandP,
+      timerP,
+      E(timerP).getCurrentTimestamp(),
+    ]);
+
+    const quoteAmount = harden({
+      brand: quoteBrand,
+      value: [
+        {
+          amountIn,
+          amountOut,
+          timer,
+          timestamp,
+        },
+      ],
+    });
+    const quotePayment = await E(quoteMint).mintPayment({
+      brand: quoteBrand,
+      value: [quoteAmount],
+    });
+    return harden({ quoteAmount, quotePayment });
```

Comment:
dupes https://github.com/Agoric/agoric-sdk/blob/f1a6fe1bfd2aa118a306ffad2608f9e6b1a79de3/packages/zoe/src/contractSupport/priceAuthorityTransform.js#L87-L95

consider DRYing. fine to have priceAuthorityInitial depend on priceAuthorityTransform so the latter could just export that maker function

Reply:
LGTM, especially the `mintQuote` naming